pkg/auth/login_credentials_manager.go

```diff
@@ -75,6 +77,8 @@ type LoginCredentialsManager interface {
 	GetCredentialsFromConfig(*config.Config, netrc.NetrcMachineParams) func() (*Credentials, error)
 	GetCredentialsFromKeychain(bool, string, string) func() (*Credentials, error)
 	GetCredentialsFromNetrc(netrc.NetrcMachineParams) func() (*Credentials, error)
+	GetOnPremSsoCredentials(url, caCertPath string, unsafeTrace bool) func() (*Credentials, error)
+	GetOnPremSsoCredentialsFromConfig(*config.Config, bool) func() (*Credentials, error)
```

Comment:
I think these functions should be able to be combined into one, right?

Reply:
Probably, yeah, I'll give it a try.

Reply:
Not sure how to do this neatly, actually.

The issue is that when you first do `confluent login --url ...`, the context is empty so you need to pass in the url and ca-cert-path values from the flag. But afterwards, when using the refresh token, I need to use the url and ca-cert-path from the Context (because some on-prem commands don't support those flags).

So a combined function needs to pass in url, ca-cert-path, and config/context (which I also need for auth and refresh token in addition to url and ca-cert-path) to account for both cases.